src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_CONSTANT = "supply";
+    private static final String BUY_CONSTANT = "buy";
+    private static final String RESULT_CONSTANT = "result";
+    private static final String WORD_SPLIT_REGEX = "\\W+";
+    private static final String SEPARATOR = " ";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFromFile(fromFileName);
+        String report = generateReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    private String generateReport(String data) {
+        String[] words = data.split(WORD_SPLIT_REGEX);
+        int supplySum = 0;
+        int buySum = 0;
+
+        for (int i = 0; i < words.length; i += 2) {
+            if (words[i].equals(SUPPLY_CONSTANT)) {
+                supplySum += Integer.parseInt(words[i + 1]);
+            } else if (words[i].equals(BUY_CONSTANT)) {
+                buySum += Integer.parseInt(words[i + 1]);
+            }
+        }
```

Comment:
this is a fragile approach, a more robust one would be to have constants for indexes in the array by which you can get values, OPERATION_INDEX, AMOUNT_INDEX, but overall it's ok as is